### ISSUE TITLE
Can't use sha references in FROM?

### ISSUE THREAD

**Issue by bparees** [GoogleContainerTools/kaniko#136]
When building a Dockerfile that uses a FROM reference that is an image id(e.g. `FROM docker.io/centos/nodejs-8-centos7@sha256:8d144d0e2a7d57b537e83f68a6a4aef495850788c6dd5173d1c75fbd45e5ca0b`), the push ultimately hits a nil pointer error while trying to handle the following error:

```
Manifest does not match provided manifest digest sha256:ac0b52de1f8370cdecf4cd8e8d422aa0e306620f3f2c5e69be76c5b7a3e748a3
```

The error seems to come back from "isMultiImage" but when it then triggers the invocation of transports.ImageName the end result is a nil pointer.
https://github.com/GoogleContainerTools/kaniko/blob/master/vendor/github.com/containers/image/copy/copy.go#L156

The same build works fine if I just specify `FROM docker.io/centos/nodejs-8-centos7`

**Comment by dlorenc** [comment]
I think this will be fixed with #140 but we should add an integration test for this as well.

**Comment by priyawadhwa** [comment]
This should be working now, so I'll close this issue!
